src/redux/actions/providers.js

```diff
@@ -82,17 +82,20 @@ export const updateSource = (source, formData, title, description) => (dispatch)
         payload: error
     }));
 
-export const removeSource = (sourceId, title) => (dispatch) =>
-    doRemoveSource(sourceId).then(_finished => dispatch({
-        type: ADD_NOTIFICATION,
-        payload: {
-            variant: 'success',
-            title
+export const removeSource = (sourceId, title) => ({
+    type: ACTION_TYPES.REMOVE_SOURCE,
+    payload: () => doRemoveSource(sourceId),
```

Comment:
The promise middleware was deliberately removed in earlier versions of this repo. Are you sure you want it Back?

Reply:
Was there any specific reason? I think it serves its purpose, the code is clean and is officially supported by notifications middleware.

Reply:
It was getting in a way for more complex requests. That is it. You can still use the notification middleware without promise middleware.